re: issue #1170, Refilling lockpick rings

b-Y-d opened the pull request:
- Removed the **ensure_copper_on_hand** helper from **pick.lic** and put it inside **refill_lockpick_container**.
- Added an optional **cost** (defaults to 125) variable to **refill_lockpick_container**.
- pick.lic now passes the lockpick cost to **refill_lockpick_container**
- **refill_lockpick_container** determines how much coin to withdraw based on the appraisal regex. **count** * **cost**.

rpherbig commented on common-items.lic:
```diff
@@ -80,13 +80,14 @@ module DRCI
 
     room = get_data('town')[hometown]['locksmithing']['id']
     count = Regexp.last_match(1).to_i
+    ensure_copper_on_hand(cost * count, hometown)
```
You will need to use the `DRCM` prefix here, and update the include and custom_require lists above.

rpherbig replied:
Rather than introducing a new dependency, you could split `refill_lockpick_container` into two methods, one to count your ring and one to refill.

rpherbig commented on common-items.lic:
```diff
@@ -80,13 +80,14 @@ module DRCI
   end
 
-  def refill_lockpick_container(lockpick_type, hometown, container)
+  def refill_lockpick_container(lockpick_type, hometown, container, cost = 125)
```
I'm not sure I like having a default value here... but if we do, I suggest erring on the side of the most expensive lockpick type.

b-Y-d commented:
- Split **refill_lockpick_container** into itself and **count_lockpick_container***
- Removed the cost variable from **refill_lockpick_container**

Need to test, currently untested!

b-Y-d commented:
Tested, found and fixed one error. Looks like it works fine.
